spyderlib/widgets/sourcecode/codeeditor.py

```diff
@@ -8,54 +8,57 @@
 Editor widget based on QtGui.QPlainTextEdit
 """
 
+# TODO: Try to separate this module from spyderlib to create a self
+#       consistent editor module (Qt source code and shell widgets library)
+
 # %% This line is for cell execution testing
 # pylint: disable=C0103
 # pylint: disable=R0903
 # pylint: disable=R0911
 # pylint: disable=R0201
 
+# Standard library imports
 from __future__ import division
-
-import sys
+from unicodedata import category
+import os.path as osp
 import re
 import sre_constants
-import os.path as osp
+import sys
 import time
-from unicodedata import category
 
-from spyderlib.qt import is_pyqt46
-from spyderlib.qt.QtGui import (QColor, QMenu, QApplication, QSplitter, QFont,
-                                QTextEdit, QTextFormat, QPainter, QTextCursor,
-                                QBrush, QTextDocument, QTextCharFormat,
-                                QPrinter, QToolTip, QCursor, QLabel,
-                                QInputDialog, QTextBlockUserData, QLineEdit,
-                                QKeySequence, QVBoxLayout, QHBoxLayout,
-                                QDialog, QIntValidator, QDialogButtonBox,
-                                QGridLayout, QPaintEvent, QMessageBox, QWidget,
-                                QTextOption)
-from spyderlib.qt.QtCore import (Qt, Signal, QTimer, QRect, QRegExp, QSize,
-                                 Slot)
-from spyderlib.qt.compat import to_qvariant
-import spyderlib.utils.icon_manager as ima
+# Third party imports
+from qtpy import is_pyqt46
+from qtpy.compat import to_qvariant
+from qtpy.QtCore import QRect, QRegExp, QSize, Qt, QTimer, Signal, Slot
+from qtpy.QtGui import (QBrush, QColor, QCursor, QFont, #QIntValidator,  # FIXME: QTPY needs to be fixed!
+                        QKeySequence, QPaintEvent, QPainter,
+                        QTextBlockUserData, QTextCharFormat, QTextCursor,
+                        QTextDocument, QTextFormat, QTextOption)
+from qtpy.QtPrintSupport import QPrinter
+from qtpy.QtWidgets import (QApplication, QDialog, QDialogButtonBox,
+                            QGridLayout, QHBoxLayout, QInputDialog, QLabel,
+                            QLineEdit, QMenu, QMessageBox, QSplitter,
+                            QTextEdit, QToolTip, QVBoxLayout, QWidget,  # )
+                            QIntValidator)  # FIXME: Remove line after QtPy fix

```

Comment:
@ccordoba12 please remove QIntValidator from the imports (delete line) after new QtPy is available.